stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/plugin/factory/StreamOfflinePluginFactory.kt

```diff
@@ -45,17 +43,7 @@ public class StreamOfflinePluginFactory(
     }
 
     private fun createOfflinePlugin(): OfflinePlugin {
-        // This can only be called after ChatClient was instantiated!!
-        val chatClient = ChatClient.instance()
-        ChatDomain.Builder(appContext, chatClient).apply {
-            if (config.backgroundSyncEnabled) enableBackgroundSync() else disableBackgroundSync()
-            if (config.persistenceEnabled) offlineEnabled() else offlineDisabled()
-            if (config.userPresence) userPresenceEnabled() else userPresenceDisabled()
-            recoveryEnabled()
-        }.build()
-
-        val userStateFlow = MutableStateFlow<User?>(null)
-        chatClient.preSetUserListeners.add { user -> userStateFlow.value = user }
+        val userStateFlow = MutableStateFlow(ChatClient.instance().getCurrentUser())
```

Comment:
Can we add a note somewhere to KDocs that there is a guarantee the plugin won't be created before current user is set?